File: backend/dto/message.py
from pydantic import BaseModel

# /api/message
# GET
class GetMessageRequest(BaseModel):
    sender_id: uuid.UUID
    recipient_id: uuid.UUID


class GetMessageResponse(BaseModel):
    class InternalMessage(BaseModel):
        message_id: uuid.UUID
        time_stamp: int
        message: str

    messages: list[InternalMessage]

# POST
class PostMessageRequest(BaseModel):
    sender_id: uuid.UUID
    recipient_id: uuid.UUID
    message: str

class PostMessageResponse(BaseModel):
    success: bool

File: frontend/src/dto/message.ts
export interface Message {
    message_id: string;
    sender_id: string;
    recipient_id: string;
    time_stamp: Date;
    message: string;
}

export interface GetMessageResponse {
    messages: Array<Message>;
}

export interface PostMessageRequest {
    recipient_id: string;
    message: string;
}

export interface PostMessageResponse {
    success: boolean;
}